### PR TITLE
linuxPackages.evdi: 2021-04-01 -> 2021-06-11, fix 5.13 build

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
Fix 5.13 build

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
@ofborg build linuxPackages_5_13.evdi

**Comment by eyJhb** [comment]
I can test this Friday I think, is that OK?

On 30 June 2021 18:06:53 CEST, "ofborg[bot]" ***@***.***> wrote:
***@***.***[bot] requested your review on: NixOS/nixpkgs#128790
>linuxPackages.evdi: 2021-04-01 -&gt; 2021-06-11, fix 5.13 build.
>
>-- 
>You are receiving this because your review was requested.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/128790#event-4960742934

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by fortuneteller2k** [comment]
> I can test this Friday I think, is that OK?
> 
> On 30 June 2021 18:06:53 CEST, "ofborg[bot]" ***@***.***> wrote:
> ***@***.***[bot] requested your review on: NixOS/nixpkgs#128790
> >linuxPackages.evdi: 2021-04-01 -&gt; 2021-06-11, fix 5.13 build.
> >
> >-- 
> >You are receiving this because your review was requested.
> >Reply to this email directly or view it on GitHub:
> >https://github.com/NixOS/nixpkgs/pull/128790#event-4960742934
> 
> -- 
> Sent from my Android device with K-9 Mail. Please excuse my brevity.

Fine by me for you to test this when you're available. :)

File: pkgs/os-specific/linux/evdi/default.nix [hunk 1]
```diff
@@ -2,15 +2,23 @@
...
-    sha256 = "09apbvdc78bbqzja9z3b1wrwmqkv3k7cn3lll5gsskxjnqbhxk9y";
+    rev = "65e12fca334f2f42396f4e8d16592d53cab34dd6";
+    sha256 = "sha256-81IfdYKadKT7vRdkmxzfGo4KHa4UJ8uJ0K6djQCr22U=";
   };
 
+  patches = [
+    # Linux 5.13 support
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/DisplayLink/evdi/pull/284.patch";
+      sha256 = "sha256-IjgYhwTG7UYrSN2ZqfTLs27G5n8ZZY5LEH27fF/bF/E=";
```
Comment (jonringer) [inline pkgs/os-specific/linux/evdi/default.nix:18]: this isn't a stable url they are able to push changes. please use a stable url. E.g.
```
https://github.com/DisplayLink/evdi/pull/284/commits/c5f5441d0a115d2cfc8125b8bafaa05b2edc7938.patch
```

**Comment by eyJhb** [comment]
I am unsure about the commits being pulled in from the PR. I am not sure what the convention is, but it seems like that might be a overlay until it is merged?

@jonringer any help here?

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 128790` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.evdi</li>
<li>linuxPackages_4_14.evdi</li>
<li>linuxPackages_4_4.evdi</li>
<li>linuxPackages_4_9.evdi</li>
<li>linuxPackages_latest-libre.evdi</li>
</ul>
</details>
<details>

  <summary>10 packages failed to build and are new build failures:</summary>
  <ul>
<li>displaylink: log was empty</li>
<li>linuxPackages.evdi: log was empty</li>
<li>linuxPackages_4_19.evdi: log was empty</li>
<li>linuxPackages_5_12.evdi: log was empty</li>
<li>linuxPackages_5_4.evdi: log was empty</li>
<li>linuxPackages_hardened.evdi: log was empty</li>
<li>linuxPackages_latest_hardened.evdi: log was empty</li>
<li>linuxPackages_lqx.evdi: log was empty</li>
<li>linuxPackages_testing_bcachefs.evdi: log was empty</li>
<li>linuxPackages_zen.evdi: log was empty</li>
  </ul>
</details>
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>linuxPackages_5_13.evdi</li>
<li>linuxPackages_xanmod.evdi</li>
</ul>
</details>

**Comment by fortuneteller2k** [comment]
@ofborg build linuxPackages_5_13.evdi

**Comment by fortuneteller2k** [comment]
Module should now build for older kernels, only applied those patches if it's building on 5.13 and newer kernels.

**Comment by eyJhb** [comment]
> Module should now build for older kernels, only applied those patches if it's building on 5.13 and newer kernels.

Testing in a moment, rebuilding

**Comment by eyJhb** [comment]
Works fine on my end with a `nixos-rebuild svitch`. :)

**Pull request merged**